src/java/com/puppetlabs/http/client/impl/ExceptionInsertingPipedInputStream.java

```diff
@@ -0,0 +1,43 @@
+package com.puppetlabs.http.client.impl;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+
+public class ExceptionInsertingPipedInputStream extends PipedInputStream {
```

Comment:
Would it make sense to also override the `close` method and do a `checkFinalResult` call within it?